src/Altinn.Profile.Integrations/Repositories/RegistrySyncMetadataRepository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Altinn.Profile.Integrations.Persistence;
+using Microsoft.EntityFrameworkCore;
+
+namespace Altinn.Profile.Integrations.Repositories;
+
+/// <summary>
+/// Defines a repository for operations related to registry sync metadata.
+/// </summary>
+public class RegistrySyncMetadataRepository(IDbContextFactory<ProfileDbContext> contextFactory) : IRegistrySyncMetadataRepository
+{
+    private readonly IDbContextFactory<ProfileDbContext> _contextFactory = contextFactory;
+
+    /// <inheritdoc />
+    public async Task<DateTime> GetLatestSyncTimestampAsync()
+    {
+        using ProfileDbContext databaseContext = await _contextFactory.CreateDbContextAsync();
+
+        var lastSync = await databaseContext.RegistrySyncMetadata.FirstOrDefaultAsync();
+        if (lastSync == null)
+        {
+            return DateTime.MinValue;
+        }
+
+        return lastSync.LastChangedDateTime;
+    }
+
+    /// <inheritdoc />
+    public async Task<DateTime> UpdateLatestChangeTimestampAsync(DateTime updated)
+    {
+        using ProfileDbContext databaseContext = await _contextFactory.CreateDbContextAsync();
+        var lastSync = await databaseContext.RegistrySyncMetadata.FirstOrDefaultAsync();
+        if (lastSync == null)
+        {
+            lastSync = new Entities.RegistrySyncMetadata
+            {
+                LastChangedId = Guid.NewGuid().ToString("N"),
```

Comment:
Do we need this be come from "registry"? `LastChangedId` sounds like an id from the feed.

Reply:
I don't think so. It is not changed in the A2 stored procedures for this table